radiopadre/textfile.py

```diff
@@ -121,60 +135,62 @@ def render_text(self, head=None, tail=None, full=None, grep=None, slicer=None, s
+            collapsed = False
+
+        title_html = self._header_html(title=title, subtitle=subtitle)
+        content_html = ""
+
         # empty head and tail: return just the title
         if not head and not tail:
-            return txt
-
-        # render as table
-        border_style = "1px solid black"
-
-        def render_line(line_num, line, firstlast, fs):
-            border_top = border_style if line_num == firstlast[0] else "none"
-            border_bottom = border_style if line_num == firstlast[1] else "none"
-            border_rl = border_style if line_num != "..." else "none"
-            background = "#f2f2f2" if line_num != "..." else "none"
-            line_num_html = ""
-            if self._show_numbers:
-                line_num_html = \
-                    """<DIV style="display: table-cell; border-top: {border_top}; border-bottom: {border_bottom};
-                                    border-left: {border_rl}; padding-left: 4px; padding-right: 4px;
-                                    height=1em; background-color: {background}">{line_num}
-                     </DIV>""".format(**locals())
-
-            return """
-                <DIV style="display: table-row; height=1em">
-                    {line_num_html}
-                    <DIV style="display: table-cell; border-top: {border_top}; border-bottom: {border_bottom};
-                                height=1em; border-right: {border_rl}; padding-left: 4px; padding-right: 4px"><PRE>{line}</PRE>
-                    </DIV>
-                </DIV>\n""".format(**locals())
-
-        lh = fs*1.5
-        txt += """<DIV style="display: table; width: 100%; font-size: {fs}em; line-height: {lh}em">""".format(**locals())
-        firstlast = self._lines[0][0], self._lines[-1][0]
-        for line_num, line in head:
-            txt += render_line(line_num, htmlize(line), firstlast, fs=fs)
-        if tail:
-            txt += render_line("&#8943;", "", firstlast, fs=fs)
-            for line_num, line in tail:
-                txt += render_line(line_num, htmlize(line), firstlast, fs=fs)
-        txt += "\n</DIV>\n"
-        return txt
+            collapsed = None
+        else:
+            show_numbers = self._show_numbers if number is None else number
+
+            # render as table
+            border_style = "1px solid black"
+
+            def render_line(line_num, line, firstlast, fs):
+                border_top = border_style if line_num == firstlast[0] else "none"
+                border_bottom = border_style if line_num == firstlast[1] else "none"
+                border_rl = border_style if line_num != "..." else "none"
+                background = "#f2f2f2" if line_num != "..." else "none"
+                line_num_html = ""
+                if show_numbers:
+                    line_num_html = \
+                        """<DIV style="display: table-cell; border-top: {border_top}; border-bottom: {border_bottom};
```

Comment:
Just out of curiosity, is there a reason why the HTML tags are in caps?

Reply:
It's just a matter of style. (Which, as you can see though, is not entirely consistent...)